src/Facility.Core/Http/HttpClientService.cs

```diff
@@ -158,6 +188,156 @@ protected HttpClientService(HttpClientServiceSettings? settings, Uri? defaultBas
 			// error contacting service
 			return ServiceResult.Failure(CreateErrorFromException(exception));
 		}
+		finally
+		{
+			httpResponse?.Dispose();
+			httpRequest?.Dispose();
+		}
+	}
+
+	/// <summary>
+	/// Sends an HTTP request for an event and processes the response.
+	/// </summary>
+	protected async Task<ServiceResult<IAsyncEnumerable<ServiceResult<TResponse>>>> TrySendEventRequestAsync<TRequest, TResponse>(HttpMethodMapping<TRequest, TResponse> mapping, TRequest request, CancellationToken cancellationToken)
+		where TRequest : ServiceDto, new()
+		where TResponse : ServiceDto, new()
+	{
+		if (mapping == null)
+			throw new ArgumentNullException(nameof(mapping));
+		if (request == null)
+			throw new ArgumentNullException(nameof(request));
+
+		HttpRequestMessage? httpRequest = null;
+		HttpResponseMessage? httpResponse = null;
+		try
+		{
+			var startRequestResult = await TryStartRequestAsync(mapping, request, cancellationToken).ConfigureAwait(false);
+			if (startRequestResult.IsFailure)
+				return startRequestResult.ToFailure();
+			(httpRequest, httpResponse, var responseMapping) = startRequestResult.Value;
+
+			var enumerable = CreateAsyncEnumerable(httpRequest, httpResponse, responseMapping, ContentSerializer, m_skipResponseValidation, cancellationToken);
+			httpResponse = null;
+			httpRequest = null;
+
+			return ServiceResult.Success(enumerable);
+		}
+		catch (OperationCanceledException) when (!cancellationToken.IsCancellationRequested)
+		{
+			// HttpClient timeout
+			return ServiceResult.Failure(ServiceErrors.CreateTimeout());
+		}
+		catch (Exception exception) when (ShouldCreateErrorFromException(exception))
+		{
+			// cancellation can cause the wrong exception
+			cancellationToken.ThrowIfCancellationRequested();
+
+			// error contacting service
+			return ServiceResult.Failure(CreateErrorFromException(exception));
+		}
+		finally
+		{
+			httpResponse?.Dispose();
+			httpRequest?.Dispose();
+		}
+	}
+
+	private async IAsyncEnumerable<ServiceResult<TResponse>> CreateAsyncEnumerable<TResponse>(HttpRequestMessage httpRequest, HttpResponseMessage httpResponse, HttpResponseMapping<TResponse> responseMapping, HttpContentSerializer contentSerializer, bool skipResponseValidation, [EnumeratorCancellation] CancellationToken cancellationToken)
+		where TResponse : ServiceDto, new()
+	{
+		Stream? stream = null;
+		try
+		{
+#if NET6_0_OR_GREATER
+			stream = await httpResponse.Content.ReadAsStreamAsync(cancellationToken).ConfigureAwait(false);
+#else
+			stream = await httpResponse.Content.ReadAsStreamAsync().ConfigureAwait(false);
+#endif
```

Comment:
Could you do `await using var` vs `using var` here (and drop the code in the `finally` block)?